test/subscription_test.dart

```diff
@@ -460,30 +460,28 @@ Future<void> main([List<String>? args]) async {
     final userY = await appY.logIn(credentials);
 
     final realmX = getRealm(Configuration.flexibleSync(userX, [Task.schema]));
-
-    final pathY = path.join(temporaryDir.path, "Y.realm");
-    final realmY = getRealm(Configuration.flexibleSync(userY, [Task.schema], path: pathY));
+    final realmY = getRealm(Configuration.flexibleSync(userY, [Task.schema]));
 
     realmX.subscriptions.update((mutableSubscriptions) {
       mutableSubscriptions.add(realmX.all<Task>());
+    });
+
+    final objectId = ObjectId();
+    realmX.write(() => realmX.add(Task(objectId)));
+
+    realmY.subscriptions.update((mutableSubscriptions) {
       mutableSubscriptions.add(realmY.all<Task>());
     });
 
     await realmX.subscriptions.waitForSynchronization();
     await realmY.subscriptions.waitForSynchronization();
 
-    final objectId = ObjectId();
-    realmX.write(() => realmX.add(Task(objectId)));
-    
     await realmX.syncSession.waitForUpload();
     await realmY.syncSession.waitForDownload();
 
-    waitForCondition(() {
-      final task = realmY.find<Task>(objectId);  
-      return task != null;
-    }, timeout: Duration(seconds: 30), retryDelay: Duration(microseconds: 500));
+    realmCore.realmRefresh(realmY);
```

Comment:
Do you suggest we add an explicit refresh to the end of `waitForDownload` or open a bug report with core? @fealebenpae argued that we should not do either.

Reply:
I wonder why this has always worked for .NET then - I just checked and we don't have explicit refreshes at `waitForDownload`.

Reply:
Likely because we have a different scheduler implementation and also because we don't need to dispatch the `wait_for_download` callback on a specific thread - awaiting it already takes care of that. It's very timing-sensitive, and the async machinery here is just different.

Reply:
this was working before but recently got worse. So what @nielsenko is doing here is the only stable fix for it.

Reply:
@nirinchev we discussed that but decided against doing it automatically since it will hurt performance.

Reply:
Technically we can't even do it automatically since we don't have easy access to the realm instance from the sync session.

Reply:
We can get any a realm instance from the coordinator and refresh that (obviously for the correct path and scheduler). Generally speaking, `realm.refresh` is a clutch and under normal circumstances people should never have to use it. The fact that `waitForDownload` doesn't behave as one would expect is kind of unfortunate but we can probably improve things later - probably once we have [`refreshAsync` implemented in Core](https://github.com/realm/realm-core/issues/5246). Can we file a ticket to make sure we don't forget though?

Reply:
I have opened https://github.com/realm/realm-core/issues/5543

Reply:
I think we need to handle this in the SDK. I don't think this is a problem with Core. The API works as intended there.

@nirinchev the realm coordinator doesn't track individual Realm instances - there can be many Realm instances (on the same or different threads) using the same coordinator.